specification/elasticsan/resource-manager/Microsoft.ElasticSan/preview/2024-06-01-preview/elasticsan.json

```diff
@@ -1069,6 +1144,99 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ElasticSan/elasticSans/{elasticSanName}/deletedVolumes": {
+      "get": {
+        "tags": [
+          "Deleted Volumes List"
+        ],
+        "description": "List Deleted Volumes.",
+        "operationId": "DeletedVolumes_ListByElasticSan",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ElasticSanNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- Operation completed successfully; retrieved deleted Volume List.",
+            "schema": {
+              "$ref": "#/definitions/VolumeList"
+            }
+          },
+          "default": {
+            "description": "Error from the ElasticSan Resource Provider.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "DeletedVolumes_ListByElasticSan_MaximumSet_Gen": {
+            "$ref": "./examples/DeletedVolumes_ListByElasticSan_MaximumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ElasticSan/elasticSans/{elasticSanName}/deletedVolumes/{volumeName}": {
+      "get": {
+        "tags": [
+          "Deleted Volumes List By Name"
```

Comment:
Is this for getting different versions of soft-deleted volumes? I thought we support it for internal soft delete only and not external soft-delete.